lib/mayaUsdAPI/python/wrapProxyShapeNotice.cpp

```diff
@@ -0,0 +1,122 @@
+//
+// Copyright 2023 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <mayaUsdAPI/proxyShapeNotice.h>
+
+#include <pxr/base/tf/instantiateSingleton.h>
+#include <pxr/base/tf/instantiateType.h>
+
+#include <boost/python/def.hpp>
+
+PXR_NAMESPACE_OPEN_SCOPE
+
+// Listen to mayaUsdAPI stage notices
+class StageNoticesListener : public TfWeakBase
+{
+public:
+    StageNoticesListener()
+    {
+        TfSingleton<StageNoticesListener>::SetInstanceConstructed(*this);
+        TfWeakPtr<StageNoticesListener> ptr(this);
+        _stageSetKey = TfNotice::Register(ptr, &StageNoticesListener::_StageSet);
+        _stageInvalidateKey = TfNotice::Register(ptr, &StageNoticesListener::_StageInvalidate);
+        _stageObjectsChangedKey
+            = TfNotice::Register(ptr, &StageNoticesListener::_StageObjectsChanged);
+        TF_VERIFY(
+            _stageSetKey.IsValid() && _stageInvalidateKey.IsValid()
+            && _stageObjectsChangedKey.IsValid());
+    }
+
+    ~StageNoticesListener() { _Revoke(); }
+
+    // StageSet notice
+    bool StageSetHasBeenCalled() const { return _stageSetCalled; }
```

Comment:
I know it's an implementation file, but StageSetHasBeenCalled() is not much clearer to me than just StageSet(), and is longer.  It's a minor point, admittedly.